tests/requirements.in

```diff
@@ -0,0 +1,5 @@
+pytest >=3.6.4,!=3.7.0,!=3.7.1,!=3.7.2,!=3.7.3,!=3.7.4,!=3.8.0,!=3.8.1 # https://github.com/pytest-dev/pytest-bdd/issues/250
+pytest-bdd >=3.1.0
+testinfra >=2.0.0
+paramiko >=2.4.2 # testinfra backend
+cryptography==2.4.2 # need this version to avoid the deprecation warning
```

Comment:
Which deprecation warning? Do we care about such warning?

Reply:
```
tests/post/steps/test_liveness.py::test_pods[paramiko://bootstrap]
  /metalk8s/.tox/tests/lib/python3.5/site-packages/paramiko/kex_ecdh_nist.py:39: CryptographyDeprecationWarning: encode_point has been deprecated on EllipticCurvePublicNumbers and will be removed in a future version. Please use EllipticCurvePublicKey.public_bytes to obtain both compressed and uncompressed point encoding.
    m.add_string(self.Q_C.public_numbers().encode_point())

tests/post/steps/test_liveness.py::test_pods[paramiko://bootstrap]
/metalk8s/.tox/tests/lib/python3.5/site-packages/paramiko/kex_ecdh_nist.py:96: CryptographyDeprecationWarning: Support for unsafe construction of public numbers from encoded data will be removed in a future version. Please use EllipticCurvePublicKey.from_encoded_point
    self.curve, Q_S_bytes

tests/post/steps/test_liveness.py::test_pods[paramiko://bootstrap]
/metalk8s/.tox/tests/lib/python3.5/site-packages/paramiko/kex_ecdh_nist.py:111: CryptographyDeprecationWarning: encode_point has been deprecated on EllipticCurvePublicNumbers and will be removed in a future version. Please use EllipticCurvePublicKey.public_bytes to obtain both compressed and uncompressed point encoding.
    hm.add_string(self.Q_C.public_numbers().encode_point())

-- Docs: https://docs.pytest.org/en/latest/warnings.html
```

Reply:
Having an exact bound, which we may need to bump later anyway (GitHub sends those 'security issues in your dependencies' emails once in a while, and I seem to remember `cryptography` has been involved...), is a bit of a  pain. I don't think these warnings are something to avoid, in the end, indeed, our deps using `cryptography` will need to get updated at some point in time.

If for now the version `pip-compile` picks when we set `cryptography >= 2.4.2` works, then all is good (even if it renders some warning, they don't do any harm). If at some point we update the deps, and a newer version of `cryptography` is picked, and the tests fail because of it, then we need to either fix Paramiko, or apply an upper bound at that point in time.

Reply:
agree

Reply:
On the other hand, from having done a bit of `behave` work recently, these warnings are a giant pain when tests fail and you need to mentally filters `cryptography` warnings from the logs you get from the test failure. Maybe we can ignore logs from a certain origin in the `pytest-bdd` configuration?